TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -48,6 +48,7 @@ public override bool BeginNode(Node node)
             RedefinesChecker<CodeElement>.OnNode(node);
             FunctionDeclarationChecker<CodeElement>.OnNode(node);
             FunctionCallChecker.OnNode(node);
+            SetStatementChecker.OnNode(node);
```

Comment:
You can override method Visit(SetStatementForIndexes) and call SetStatementChecker from here.
This will avoid a cast.